lib/project_metric_slack.rb

```diff
@@ -55,19 +55,23 @@ def compute_member_hex_colors_for_heatmap normalized_member_scores
   end
 
   def get_slack_message_totals
-    start_time = (Time.now - (7+Time.now.wday+1).days).to_s[0, 10]
-    end_time = (Time.now - (Time.now.wday).days).to_s[0, 10]
-    date_range = "after:#{start_time} before:#{end_time}"
-    get_member_names_for_channel.inject({}) do |slack_message_totals, user_name|
-      messages = @client.search_all(query: "from:#{user_name} in:#{@channel} #{date_range}").messages
-      slack_message_totals.merge user_name => messages.total
+    start_date = (Time.now - (7+Time.now.wday+1).days).to_date
+    end_date = (Time.now - (Time.now.wday).days).to_date
+    member_names_by_id = get_member_names_by_id
+    id = @client.channels_list['channels'].detect { |c| c['name'] == @channel }.id
+    history = @client.channels_history(channel: id)
+    history.messages.inject(Hash.new(0)) do |slack_message_totals, message|
+      add_to_total = 0
+      add_to_total = 1 if start_date < Time.at(message.ts.to_i).to_date && Time.at(message.ts.to_i).to_date < end_date
+      slack_message_totals.merge member_names_by_id[message.user] => slack_message_totals[message.user] + add_to_total
```

Comment:
There is a bug in the logic here.  I found it while working on slack_trend_metrics and trying to reconcile per day totals versus totals per week.  This line should be something like:
      slack_message_totals.merge member_names_by_id[message.user] => (slack_message_totals[member_names_by_id[message.user]]||0) + add_to_total
```
notice the additional indexing into member_names_by_id inside the merge call